okhttp/src/main/kotlin/okhttp3/OkHttpClient.kt

```diff
@@ -1066,6 +1099,42 @@ open class OkHttpClient internal constructor(
       this.minWebSocketMessageToCompress = bytes
     }
 
+    /**
+     * Configures OkHttp to not authenticate the HTTPS server on to [hostName]. This makes the
+     * client vulnerable to man-in-the-middle attacks and should only be used only in private
+     * development environments and only to carry test data.
+     *
+     * The following protections are removed:
+     * - The server’s TLS certificate **does not need to be signed** by a trusted certificate
+     * authority. Instead, OkHttp will trust any well-formed certificate, even if it is self-signed.
+     * This is necessary for testing against localhost, or in development environments where a
+     * certificate authority is not possible.
+     * - The server’s TLS certificate **does not need to match** the requested hostname. For
+     * example, if the certificate is issued to `example.com` and the request is to `localhost`,
+     * the certificate will still be accepted!
+     *
+     * Other TLS features are still used but provide no security benefits in absence of the above
+     * gaps. For example, an insecure TLS connection is capable of negotiating HTTP/2 with ALPN and
+     * it also has a regular-looking handshake.
+     *
+     * n.b. only available from Android version 24, due to use of host specific certificate checks
+     * added in Android.
+     *
+     * @param hostname the exact hostname from the URL for insecure connections.
+     */
+    fun insecureForHost(hostname: String): Builder = apply {
+      val override = TrustManagerOverride(hostname, OkHostnameVerifier.INSECURE, InsecureTrustManager)
+
+      val overrides = trustManagerOverrides ?: mutableListOf<TrustManagerOverride>().also {
+        this.trustManagerOverrides = it
+      }
+
+      overrides.add(override)
+
+      // invalidate and rebuild sslSocketFactory later
+      this.sslSocketFactoryOrNull = null
```

Comment:
I don’t like how it silently throws away your configured sslSocketFactory if you call these methods in a different order